docs/CANBus.md

```diff
@@ -0,0 +1,391 @@
+  used to duplicate a CAN interface in the device model so it can be
+  passed-through to more than one Edge Application
+
+Only the following parameters are mandatory:
+
+* ztype
+
+* phylabel
+
+* logicallabel
+
+* cbattr (at least bitrate should be present for IO_TYPE_CAN type)
+
+Parameters not supported by the CAN interface shall be omitted from the device model file.
+
+### Giving access to host's CAN controllers to Edge Applications
+
+It’s important to point out that in case of CAN interfaces, the same CAN
+interface can be assigned to one or more Edge Applications. This
+requirement breaks the logic for device passthrough on EVE + Cloud
+Controller. So the alternative approach is to use the logical CAN
+definition, which points to a physical/virtual CAN interface in the device
+model. For instance, the use case from Fig. 2 would have the following
+configuration in the device model:
+
+```json
+"ioMemberList": [
+    {
+        "ztype": "IO_TYPE_CAN",
+        "phylabel": "can0",
+        "logicallabel": "can0",
+        "assigngrp" : "",
+        "phyaddrs" : {
+            "ifname" : "can0"
+        },
+        "cbattr" : {
+            "bitrate": "125000",
+        }
+    },
+    {
+        "ztype": "IO_TYPE_LCAN",
+        "phylabel": "can0.1",
+        "logicallabel": "can0.1",
+        "assigngrp" : "can0.1",
+        "phyaddrs" : {
+            "ifname" : "can0"
+        },
+    },
+    {
+        "ztype": "IO_TYPE_LCAN",
+        "phylabel": "can0.2",
+        "logicallabel": "can0.2",
+        "assigngrp" : "can0.2",
+        "phyaddrs" : {
+            "ifname" : "can0"
+        },
+    },
+    {
+        "ztype": "IO_TYPE_LCAN",
+        "phylabel": "can0.3",
+        "logicallabel": "can0.3",
+        "assigngrp" : "can0.3",
+        "phyaddrs" : {
+            "ifname" : "can0"
+        },
+    }
+]
+```
+
+The correspondence between the I/O devices declared in the device model and
+all CAN interfaces + Edge Applications are illustrated in the Fig. 4.
+
+![Fig. 4: Use case 2 with I/O CAN devices from device model illustrated.](./images/eve-can-case2-model.png)
+
+*Fig. 4: Use case 2 with I/O CAN devices from device model illustrated.*
+
+So logically the device is presented with 4 CAN interfaces, the physical
+(real) controller plus 3 logical interfaces, where each one can be
+passed-through to each Edge Application. However, all of them point to the
+same physical _can0_. In the device model, the interface _can0_ is the one that
+describes the physical CAN interface and contains the proper parameters to
+set up the controller. The logical interfaces _can0.1_, _can0.2_ and _can0.3_
+point to _can0_ and can be passed-through to the applications.
+
+The use case from Fig. 3 would have the following configuration in the device
+model (illustrated by Fig. 5):
+
+```json
+"ioMemberList": [
+    {
+        "ztype": "IO_TYPE_VCAN",
+        "phylabel": "vcan0",
+        "logicallabel": "vcan0",
+        "assigngrp" : "",
```

Comment:
We should probably also document the meaning of `assigngrp` for CAN devices (e.g. that it should be different for LCANs even if they reference the same CAN).
Is it actually looked at by domaingr at all in the case of CAN devices?

Reply:
Yes, good point. I've updated the documentation.